scipy/optimize/zeros.py

```diff
@@ -62,11 +62,22 @@ def results_c(full_output, r):
         return r
 
 
+def _newton_halt(p, dp, fp, tol, halt):
+    if halt == 'astep':
+        res = True if abs(dp) <= tol else False
+    elif halt == 'rstep':
+        res = True if abs(dp) <= tol*abs(p) else False
+    elif halt == 'afval':
+        res = True if abs(fp) <= tol else False
+    else:
+        res = True if abs(fp) <= tol*abs(p) else False
```

Comment:
all these `res = True if <something> else False` could be written `res = <statement>` since `<statement>` already evaluates as a boolean value.

also this could also just be a dict like:
```
return {
    'astep': abs(dp) <= tol,
    'rstep': abs(dp) <= tol*abs(p),
    ....
}[halt]
````